replica/table.cc

```diff
@@ -2910,23 +2909,17 @@ future<> table::apply(const mutation& m, db::rp_handle&& h, db::timeout_clock::t
     }, timeout);
 }
 
-template void table::do_apply(compaction_group& cg, db::rp_handle&&, const mutation&);
-
 future<> table::apply(const frozen_mutation& m, schema_ptr m_schema, db::rp_handle&& h, db::timeout_clock::time_point timeout) {
     if (_virtual_writer) [[unlikely]] {
-        return (*_virtual_writer)(m);
+        co_await (*_virtual_writer)(m);
+        co_return;
```

Comment:
Using coroutines here will introduce an unconditional allocation on the fast-path. I think @avikivity doesn't likes those.

Reply:
absolutely, this is a very hot path, please measure with perf-simple-query --write.